modules/miner.go

```diff
@@ -14,6 +15,15 @@ type MinerInfo struct {
 	Address        types.UnlockHash
 }
 
+// MinerWork is all the information a miner needs to do work
+type MinerWork struct {
+	Block      types.Block
+	ParentID   types.BlockID
+	Nonce      uint64
+	MerkleRoot crypto.Hash
+	Target     types.Target
+}
```

Comment:
Block already has a ParentID and Nonce field, you could probably just fill those out in the block.

But I think really you should be sending a block header, which woud look like this:

```
    ParentID
    Nonce
    TransactionMerkleRoot
```

The ID is derived by hashing the three of those together, so you don't need any more information.

Then you'd return stuff to a `/postwork' api call that just has a nonce.